From 5d593107954766e516987c853c0d28881201b9b4 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Fri, 3 May 2013 13:08:26 +0800
Subject: [PORT FROM MAIN] LiveSession: Signal the mCondition when fetch playlist failed.

BZ: 65638

When seeking, the main thread is sleep in LiveSession::seekTo
function. If error encounted in fetching playlist, signal the
mCondition to wake up the main thread or else it would cause
ANR.

Change-Id: I2b5f1053827a9f80e13c869e58fc92e98f8942f5
Orig-Change-Id: Id7311a8e7678d7fa8362b21c20b7fd572a80b046
Signed-off-by: XiaoFeng <fengx.xiao@intel.com>
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/httplive/LiveSession.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/httplive/LiveSession.cpp b/media/libstagefright/httplive/LiveSession.cpp
index 505bdb3..817857e 100644
--- a/media/libstagefright/httplive/LiveSession.cpp
+++ b/media/libstagefright/httplive/LiveSession.cpp
@@ -566,6 +566,12 @@ rinse_repeat:
                 ALOGE("failed to load playlist at url '%s'", url.c_str());
                 signalEOS(ERROR_IO);
 
+                if (mSeekTimeUs >= 0) {
+                    mSeekTimeUs = -1;
+                    Mutex::Autolock autoLock(mLock);
+                    mSeekDone = true;
+                    mCondition.broadcast();
+                }
                 return;
             }
         } else {
-- 
1.7.4.1

